src/Common/ZooKeeper/ZooKeeperImpl.cpp

```diff
@@ -840,9 +840,9 @@ void ZooKeeper::finalize(bool error_send, bool error_receive)
 
     auto expire_session_if_not_expired = [&]
     {
-        /// No new requests will appear in queue after close()
-        bool was_already_closed = requests_queue.close();
-        if (!was_already_closed)
+        /// No new requests will appear in queue after finish()
+        bool was_already_finished = requests_queue.finish();
```

Comment:
We must not clear `requests_queue` here, see https://github.com/ClickHouse/ClickHouse/blob/fe6b7c77c7d6bd2a45a20f3b6bb4eb91da6177ff/src/Common/ZooKeeper/ZooKeeperImpl.cpp#L952-L955